contracts/src/v0.8/l2ep/dev/scroll/ScrollSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,242 @@
+    _setL1Sender(to);
+  }
+
+  /// @notice internal method that stores the L1 sender
+  function _setL1Sender(address to) private {
+    address from = s_l1Sender;
+    if (from != to) {
+      s_l1Sender = to;
+      emit L1SenderTransferred(from, to);
+    }
+  }
+
+  /// @dev Returns an AggregatorV2V3Interface compatible answer from status flag
+  /// @param status The status flag to convert to an aggregator-compatible answer
+  function _getStatusAnswer(bool status) private pure returns (int256) {
+    return status ? int256(1) : int256(0);
+  }
+
+  /// @notice Helper function to record a round and set the latest feed state.
+  /// @param roundId The round ID to record
+  /// @param status Sequencer status
+  /// @param timestamp The L1 block timestamp of status update
+  function _recordRound(uint80 roundId, bool status, uint64 timestamp) private {
+    uint64 updatedAt = uint64(block.timestamp);
+    Round memory nextRound = Round(status, timestamp, updatedAt);
+    FeedState memory feedState = FeedState(roundId, status, timestamp, updatedAt);
+
+    s_rounds[roundId] = nextRound;
+    s_feedState = feedState;
+
+    emit NewRound(roundId, msg.sender, timestamp);
+    emit AnswerUpdated(_getStatusAnswer(status), roundId, timestamp);
+  }
+
+  /// @notice Helper function to update when a round was last updated
+  /// @param roundId The round ID to update
+  /// @param status Sequencer status
+  function _updateRound(uint80 roundId, bool status) private {
+    uint64 updatedAt = uint64(block.timestamp);
+    s_rounds[roundId].updatedAt = updatedAt;
+    s_feedState.updatedAt = updatedAt;
+    emit RoundUpdated(_getStatusAnswer(status), updatedAt);
+  }
+
+  /// @notice Record a new status and timestamp if it has changed since the last round.
+  /// @dev This function will revert if not called from `l1Sender` via the L1->L2 messenger.
+  ///
+  /// @param status Sequencer status
+  /// @param timestamp Block timestamp of status update
+  function updateStatus(bool status, uint64 timestamp) external override {
+    FeedState memory feedState = s_feedState;
+    if (
+      msg.sender != address(s_l2CrossDomainMessenger) || s_l2CrossDomainMessenger.xDomainMessageSender() != s_l1Sender
+    ) {
+      revert InvalidSender();
+    }
+
+    // Ignore if latest recorded timestamp is newer
+    if (feedState.startedAt > timestamp) {
+      emit UpdateIgnored(feedState.latestStatus, feedState.startedAt, status, timestamp);
+      return;
+    }
+
+    if (feedState.latestStatus == status) {
+      _updateRound(feedState.latestRoundId, status);
+    } else {
+      feedState.latestRoundId += 1;
+      _recordRound(feedState.latestRoundId, status, timestamp);
+    }
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestAnswer() external view override checkAccess returns (int256) {
+    FeedState memory feedState = s_feedState;
+    return _getStatusAnswer(feedState.latestStatus);
```

Comment:
This is a strange way the original person wrote these functions, let's just say 
```
_getStatusAnswer(s_feedState.latestStatus);
```

same with all the others below, no need to assign only to use it once